### ISSUE TITLE
Trying to display buttons for text and background color in Sulu ckeditor

### ISSUE THREAD

**Issue by SarramegnaG** [sulu/sulu#3313]
Greetings everyone,

| Q | A
| --- | ---
| Bug? | no
| New Feature? | no
| Sulu Version | 1.5
| Browser Version | Chrome

#### Actual Behavior

I'm trying to display ckeditor buttons to enable text color and background color customizations.
When setting the config, the buttons for text color and background color don't appear.

#### Expected Behavior

They should appear (see https://github.com/sulu/sulu/issues/864#issuecomment-165091811).

#### Steps to Reproduce
It looks like you have files for this plugin in `vendor/sulu/sulu/src/Sulu/Bundle/AdminBundle/Resources/public/js/vendor/husky/vendor/ckeditor/plugins/colorbutton`.

`app/Resources/views/templates/homepage.html.twig`
```
<?xml version="1.0" ?>
<template xmlns="http://schemas.sulu.io/template/template"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/template-1.0.xsd">

    <key>homepage</key>

    <view>templates/homepage</view>
    <controller>SuluWebsiteBundle:Default:index</controller>
    <cacheLifetime>2400</cacheLifetime>

    <meta>
        <title lang="en">Homepage</title>
        <title lang="de">Startseite</title>
    </meta>

    <properties>
        <section name="highlight">
            <properties>
                <property name="title" type="text_line" mandatory="true">
                    <meta>
                        <title lang="en">Title</title>
                        <title lang="de">Titel</title>
                    </meta>
                    <params>
                        <param name="headline" value="true"/>
                    </params>

                    <tag name="sulu.rlp.part"/>
                </property>

                <property name="url" type="resource_locator" mandatory="true">
                    <meta>
                        <title lang="en">Resourcelocator</title>
                        <title lang="de">Adresse</title>
                    </meta>

                    <tag name="sulu.rlp"/>
                </property>
            </properties>
        </section>

        <property name="article" type="text_editor">
            <meta>
                <title lang="en">Article</title>
                <title lang="de">Artikel</title>
            </meta>

            <params>
                <param name="table" value="true"/>
                <param name="link" value="true"/>
                <param name="paste_from_word" value="true"/>
                <param name="height" value="100"/>
                <param name="max_height" value="200"/>
                <!-- CKEditor Parameters examples: -->
                **<param name="custom_config" value="plugins/colorbutton/plugin.js" />
                <param name="toolbar" value='[{"name":"colors","items":["TextColor","BGColor"]}]' />
                <param name="extra_plugins" value="justify,format,sourcearea,link,table,pastefromword,autogrow,colorbutton" />**
            </params>
        </property>
    </properties>
</template>
```

Thanks for the help :)

**Comment by alexander-schranz** [comment]
Hello this is not longer possible in the xml definition you need to register your bundle as admin bundle and include in the admin.js and use the `app.sandbox.ckeditor.addToolbarButton` function if you have problems let me know and I invest some time to write a complete example for the color chooser.

Have a look at the exist issue: https://github.com/sulu/sulu-docs/issues/320

**Comment by SarramegnaG** [comment]
Hello @alexander-schranz,

I'm kind of new with Sulu, but I managed to do something that almost works.

- I defined an admin bundle as said in these tutorials http://blog.sulu.io/how-to-develop-a-bundle-in-the-sulu-admin-1 & http://blog.sulu.io/how-to-develop-a-bundle-in-the-sulu-admin-2
- I added the lines ```
            app.sandbox.ckeditor.addToolbarButton('color', 'Color1', 'paint-brush');
            app.sandbox.ckeditor.addToolbarButton('colorbutton', 'Color2', 'paint-brush');
            app.sandbox.ckeditor.addToolbarButton('textcolor', 'Color3', 'paint-brush');```
just in case, because I don't know which one should work...
- I have the icons in the User roles text editor tab and checked them
- I have no icon in the ckeditor

Here the repo of this : https://github.com/SarramegnaG/poc-sulu

Any hints ?

**Comment by alexander-schranz** [comment]
Not sure but think we have a bug here adding ckeditor standard plugins normally I would say it should be:

```
app.sandbox.ckeditor.addPlugin('colorbutton');
app.sandbox.ckeditor.addPlugin('colordialog');

app.sandbox.ckeditor.addToolbarButton('colors', 'TextColor', 'paint-brush');
app.sandbox.ckeditor.addToolbarButton('colors', 'BGColor', 'paint-brush');
```

think there is a error in this implementation: https://github.com/massiveart/husky/blob/9ac8b5fd984b54be0c237c8158cb78058069a002/husky_extensions/ckeditor-extension.js#L175-L179

this should be:

```
addPlugin: function(name, plugin) {
    plugins.push(name);
    if (plugin) {
        CKEDITOR.plugins.add(name, plugin);
    }
},
```

@danrot @wachterjohannes or is there another way how the standard plugins should be added?

**Comment by SarramegnaG** [comment]
Thanks @alexander-schranz , looks like it works with this code plus the fix for husky.

So what's the next step ? I have to submit a PR to husky ?

**Comment by wachterjohannes** [comment]
as long our custom plugins will work with this fix (and I see no problem) this can be added to husky. for me, this can also be shipped as HOTFIX so when you create a PR in husky to master we can ship that with the next HOTFIX release in sulu.

**Comment by danrot** [comment]
I don't really understand why that change makes a difference, but it doesn't look to me as it is causing any troubles. If testing isn't revealing any, then I don't have a problem with releasing that as a hotfix.

**Comment by SarramegnaG** [comment]
Maybe use 
```javascript
addPlugin: function(name, plugin) {
    plugins.push(name);
    if (!!plugin) {
        CKEDITOR.plugins.add(name, plugin);
    }
},
```
to stay consistent with line https://github.com/massiveart/husky/blob/9ac8b5fd984b54be0c237c8158cb78058069a002/husky_extensions/ckeditor-extension.js#L188

**Comment by alexander-schranz** [comment]
@SarramegnaG yes just create a PR into the master branch of husky that we can create a hotfix release for it.

**Comment by SarramegnaG** [comment]
@alexander-schranz @wachterjohannes @danrot should be ok, that's my first PR ;)

**Comment by danrot** [comment]
Fixed in #3356
